app/pages/01_Explore.py

```diff
@@ -64,19 +140,84 @@ def _upload_to_api(file):
             n_rows = result["n_rows"]
             n_cols = result["n_cols"]
 
-            # Update session state and rerun
             st.session_state["dataset_choice"] = table_name
             st.session_state["uploader_nonce"] = nonce + 1
             st.session_state["flash"] = (
                 f"Ingested **{dataset_id}** as `{table_name}` ({n_rows}×{n_cols})."
             )
             st.rerun()
 
-    elif uploaded.name.lower().endswith(".zip"):
-        st.warning(
-            "ZIP upload not yet implemented via API. Extract and upload CSV directly."
+    elif lower_name.endswith(".zip"):
+        response = _upload_zip_to_api(uploaded)
+        if response:
+            default_name = sanitize_id(os.path.splitext(uploaded.name)[0])
+            if not default_name:
+                default_name = f"zip_{response['zip_id'][:8]}"
+            st.session_state["zip_session"] = {
+                "zip_id": response["zip_id"],
+                "files": response.get("files", []),
+                "filename": uploaded.name,
+            }
+            st.session_state["zip_dataset_name"] = default_name
+            st.session_state["uploader_nonce"] = nonce + 1
+            st.rerun()
+
+
+zip_session = st.session_state.get("zip_session") or None
+
+if zip_session:
+    st.subheader("ZIP ingestion")
+    st.caption(
+        "Select which files inside the archive should be combined and ingested into DuckDB."
+    )
+
+    files_in_zip = zip_session.get("files", [])
+    supported_files = [f for f in files_in_zip if _is_supported_zip_file(f)]
+    unsupported_files = sorted(set(files_in_zip) - set(supported_files))
+
+    if unsupported_files:
+        st.info(
+            "Unsupported files will be skipped: " + ", ".join(unsupported_files)
         )
```

Comment:
Same comment that I had for the api error handling (api/main.py line 213) but here it's probably more important since the user sees it. - I think it'd be better to just show invalid filetypes found and a count of how many times they were found in the archive, not all invalid files.